support-frontend/assets/pages/[countryGroupId]/checkout.tsx

```diff
@@ -912,20 +914,27 @@ function CheckoutComponent({
 			};
 			setErrorMessage(undefined);
 			setErrorContext(undefined);
-			const createSubscriptionResult = await fetch('/subscribe/create', {
+			const createResponse = await fetch('/subscribe/create', {
 				method: 'POST',
 				body: JSON.stringify(createSupportWorkersRequest),
 				headers: {
 					'Content-Type': 'application/json',
 				},
-			})
-				.then((response) => response.json())
-				.then((json) => json as StatusResponse);
+			});
+
+			let processPaymentResponse: ProcessPaymentResponse;
+
+			if (createResponse.ok) {
+				const statusResponse = (await createResponse.json()) as StatusResponse;
+				processPaymentResponse = await processPayment(statusResponse, geoId);
+			} else {
+				const errorReason = (await createResponse.text()) as ErrorReason;
```

Comment:
Do we always assume the error is text, ie is there a situation where an error returned could be json object instead?

Reply:
It looks like the convention on the server is to return a single string code for errors